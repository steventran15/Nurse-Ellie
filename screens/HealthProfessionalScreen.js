import React, { useState }from 'react';
import { View, Text, TextInput, TouchableOpacity, Image, Dimensions, StyleSheet, Keyboard, KeyboardAvoidingView } from 'react-native';
import * as Animatable from 'react-native-animatable';
import { firebase } from '../components/Firebase/config'
import Background from '../components/hpBackground';
import NurseEllieLogo from '../assets/images/nurse-ellie-logo.svg';
import MenuIcon from '../assets/images/hp-menu-icon.svg';
import BlueAddIcon from '../assets/images/blue-add-icon';
var screenHeight = Dimensions.get("window").height;
var screenWidth = Dimensions.get("window").width;

const HealthProfessionalScreen = ({navigation}) => {
    const [FieldofPractice, setFieldofPractice] = useState('')
    const [LicenseNumber, setLicenseNumber] = useState('')
    const [RegulatoryBody, setRegulatoryBody] = useState('')
    const usersRef = firebase.firestore().collection('users')

   const onHealthPress = async (res) => {
        const data = await firebase.auth().currentUser.uid
        var userDoc = firebase.firestore().collection("users").doc(data).update({
        'FieldofPractice':'',
        'LicenseNumber': '',
        'RegulatoryBody': ''
        })

        const obj = {
        FieldofPractice,
        LicenseNumber,
        RegulatoryBody,
        };
        
        usersRef.doc(data).update(obj)

    }

    const HealthProfAlert = () => {
        alert('Your account will be verified in the next  2-3 business days. Thank you');
        navigation.navigate('HomeScreen')
      };


    return (
        
        <View style={styles.container}>
            <Background/>
            <TouchableOpacity style={styles.menuButton} onPress={() => navigation.openDrawer()}>
                <MenuIcon />
            </TouchableOpacity>
            <Animatable.View style={styles.drawer} animation="fadeInUpBig"> 

            <View style={{ flexDirection: 'row', alignItems: 'center', marginBottom: screenHeight / 10 }}>
                <NurseEllieLogo height={75} style={{ flex: 1, marginRight: '5%' }} />
                <Text style={{ fontFamily: 'roboto-regular', fontSize: 25, }}> {`Account Change:\n Health Professional`}</Text>
            </View>

                <TextInput style={styles.textInput} placeholder="Field of Practice" autoCapitalize="none"  onChangeText={(text) => setFieldofPractice(text)}
                    value={FieldofPractice} returnKeyType='done' onSubmitEditing={Keyboard.dismiss}/>
                <TextInput style={styles.textInput}  placeholder="License Number" autoCapitalize="none" onChangeText={(text) => setLicenseNumber(text)} 
                 value={LicenseNumber} returnKeyType='done' onSubmitEditing={Keyboard.dismiss}/>
                <TextInput style={styles.textInput} placeholder="Regulatory Body" autoCapitalize="none"  onChangeText={(text) => setRegulatoryBody(text)}
                    value={RegulatoryBody} returnKeyType='done' onSubmitEditing={Keyboard.dismiss}/>
                <Text style={{marginBottom: 15}}>Status: Pending</Text>
                <TouchableOpacity style={styles.button} onPress={() => { onHealthPress(); HealthProfAlert(); }}>
                    <BlueAddIcon/>
                </TouchableOpacity>

                <View style={styles.whitePadding}/>
            </Animatable.View>
        </View>
    )
}


const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
    },
    heading: {
        flex: 1,
        justifyContent: 'flex-end',
        paddingHorizontal: 20,
        paddingBottom: 5
    },
    headerFont: {
        fontFamily: 'roboto-regular',
        fontSize: 28,
        fontWeight: "100", 
    },
    whitePadding: {
        height: screenHeight / 8
    },
    textInput: {
        borderBottomColor: 'rgba(112, 112, 112, 0.7)',
        borderBottomWidth: 1.5,
        fontSize: 16,
        paddingTop: 8
    },
    descriptionFont: {
        fontFamily: 'roboto-regular', 
        fontSize: 14, 
        textAlign: 'center',
        color: 'rgba(0, 0, 0, 0.7)'
    },
    clickableFont: {
        fontFamily: 'roboto-medium',
        fontSize: 14,
    },
    button: {
        marginTop: 30,
        alignSelf: 'flex-start'
    },
    menuButton: {
        position: 'absolute',
        right: 30,
        top: 40
    },
    drawer: {
        flex: 4,
        backgroundColor: '#fff',
        borderTopLeftRadius: 30,
        borderTopRightRadius: 30,
        paddingVertical: 50,
        paddingHorizontal: 30,
        position: 'absolute',
        width: screenWidth,
        height: screenHeight * 0.85,
        top: screenHeight * 0.15
    }
});

export default HealthProfessionalScreen;


