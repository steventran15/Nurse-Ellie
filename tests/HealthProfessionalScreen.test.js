import React from 'react';
import renderer from 'react-test-renderer';
import HealthProfessionalScreen from '../screens/HealthProfessionalScreen';


test('HealthProfessionalScreen renders correctly', ()=>{
    const wrapper = renderer.create(<HealthProfessionalScreen/>);
    expect(wrapper.toJSON()).toMatchSnapshot();
});

