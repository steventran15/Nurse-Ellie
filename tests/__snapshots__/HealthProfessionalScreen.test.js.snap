// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HealthProfessionalScreen renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#4285C8",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(51, 112, 172, 0.8)",
          "borderRadius": 250,
          "height": 500,
          "position": "absolute",
          "right": -375,
          "top": 667,
          "width": 500,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 30,
          "height": 60,
          "left": 10,
          "position": "absolute",
          "top": 30,
          "width": 60,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 125,
          "height": 250,
          "left": 150,
          "position": "absolute",
          "top": 222.33333333333334,
          "width": 250,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 300,
          "height": 600,
          "left": 300,
          "position": "absolute",
          "top": -333.5,
          "width": 600,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.3)",
          "borderRadius": 175,
          "height": 350,
          "left": -250,
          "position": "absolute",
          "top": 667,
          "width": 350,
        }
      }
    />
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 30,
        "top": 40,
      }
    }
  >
    <svg
      data-file-name="SvgHpMenuIcon"
    />
  </View>
  <View
    forwardedRef={[Function]}
    style={
      Object {
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 30,
        "borderTopRightRadius": 30,
        "flex": 4,
        "height": 1133.8999999999999,
        "opacity": 0,
        "paddingHorizontal": 30,
        "paddingVertical": 50,
        "position": "absolute",
        "top": 200.1,
        "transform": Array [
          Object {
            "translateY": 500,
          },
        ],
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 133.4,
        }
      }
    >
      <svg
        data-file-name="SvgNurseEllieLogo"
        height={75}
        style={
          Object {
            "flex": 1,
            "marginRight": "5%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "roboto-regular",
            "fontSize": 25,
          }
        }
      >
         
        Account Change:
 Health Professional
      </Text>
    </View>
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Field of Practice"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "borderBottomColor": "rgba(112, 112, 112, 0.7)",
          "borderBottomWidth": 1.5,
          "fontSize": 16,
          "paddingTop": 8,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="License Number"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "borderBottomColor": "rgba(112, 112, 112, 0.7)",
          "borderBottomWidth": 1.5,
          "fontSize": 16,
          "paddingTop": 8,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <TextInput
      allowFontScaling={true}
      autoCapitalize="none"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholder="Regulatory Body"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Object {
          "borderBottomColor": "rgba(112, 112, 112, 0.7)",
          "borderBottomWidth": 1.5,
          "fontSize": 16,
          "paddingTop": 8,
        }
      }
      underlineColorAndroid="transparent"
      value=""
    />
    <Text
      style={
        Object {
          "marginBottom": 15,
        }
      }
    >
      Status: Will be loaded from firebase db
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(0, 0, 0, 0.7)",
          "fontFamily": "roboto-regular",
          "fontSize": 14,
          "textAlign": "center",
        }
      }
    >
      Your account will be verified in the next  2-3 business days. Thank you
    </Text>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "flex-start",
          "marginTop": 30,
          "opacity": 1,
        }
      }
    >
      <svg
        data-file-name="SvgBlueAddIcon"
      />
    </View>
    <View
      style={
        Object {
          "height": 166.75,
        }
      }
    />
  </View>
</View>
`;
